tests/test_lib_utils.py

```diff
@@ -430,3 +430,11 @@ def test_17_convert_column_to_unicode(self):
         self.assertEqual(convert_column_to_unicode(False), "False")
         self.assertEqual(convert_column_to_unicode(b"yes"), u"yes")
         self.assertEqual(convert_column_to_unicode(u"yes"), u"yes")
+
+    def test_18_censor_connect_string(self):
+        self.assertEqual(censor_connect_string("mysql://pi:kW44sqqWtGYX@localhost/pi"),
+                         "mysql://pi:xxxx@localhost/pi")
+        self.assertEqual(censor_connect_string("mysql://pi:kW44s@@qqWtGYX@localhost/pi"),
+                         "mysql://pi:xxxx@localhost/pi")
+        self.assertEqual(censor_connect_string(u"mysql://knöbel:föö@localhost/pi"),
+                         u"mysql://knöbel:xxxx@localhost/pi")
```

Comment:
Does it also succeed with something like

   "psql+odbc://pi@localhost/pi"

There are URLs, that have a "+" in the schema and no passwords.

Reply:
I've added more tests for this.